cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/classloader/SparkClassRewriter.java

```diff
@@ -442,6 +453,24 @@ public void visitEnd() {
     return cw.toByteArray();
   }
 
+  private static class UnmodifiableData {
```

Comment:
Move private class to the end of the file. Not to interrupt the code flow. Also, add a javadoc for the class.

Reply:
done